diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt

```diff
@@ -358,7 +359,10 @@ class IdentifierNaming(private val configRules: List<RulesConfig>) : Rule("ident
         if (!node.isOverridden()) {
             // if function has Boolean return type in 99% of cases it is much better to name it with isXXX or hasXXX prefix
             if (functionReturnType != null && functionReturnType == PrimitiveType.BOOLEAN.typeName.asString()) {
-                if (booleanMethodPrefixes.none { functionName.text.startsWith(it) }) {
+                val configuration = BooleanFunctionsConfiguration(
+                        this.configRules.getRuleConfig(FUNCTION_BOOLEAN_PREFIX)?.configuration ?: emptyMap()
+                )
+                if (booleanMethodPrefixes.none { functionName.text.startsWith(it) } && functionName.text !in configuration.allowedBooleanFunctions) {
```

Comment:
These should be prefixes, not names. Like when you need functions `equivalentContent`, `equivalentSize` etc., you don't want to list them all. So here you should check `startsWith` too.

Reply:
Fixed